File: LeetCode/Easy/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length;
        int ans = 0;
        int minValue = prices[0];
        
        for (int i = 0; i < n; i++) {
            ans = Math.max(ans, prices[i] - minValue);
            minValue = Math.min(minValue, prices[i]);
        }

        return ans;
    }
}

File: LeetCode/Easy/0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    min := prices[0]
    profit := 0

    for i := 0; i < len(prices); i++ {
        diff := prices[i] - min 
        if (diff > profit) {
            profit = diff
        }
        if (prices[i] < min) {
            min = prices[i]
        }
    }

    return profit
}